pyomo/contrib/gdpopt/util.py

```diff
@@ -144,24 +142,37 @@ def a_logger(str_or_logger):
         return logging.getLogger(str_or_logger)
 
 
-def copy_var_list_values(from_list, to_list, config, skip_stale=False):
-    """Copy variable values from one list to another."""
+def copy_var_list_values(from_list, to_list, config,
+                         skip_stale=False, skip_fixed=True,
+                         ignore_integrality=False):
+    """Copy variable values from one list to another.
+
+    Rounds to Binary/Integer if neccessary
+    Sets to zero for NonNegativeReals if neccessary
+    """
     for v_from, v_to in zip(from_list, to_list):
         if skip_stale and v_from.stale:
             continue  # Skip stale variable values.
+        if skip_fixed and v_to.is_fixed():
+            continue  # Skip fixed variables.
         try:
             v_to.set_value(value(v_from, exception=False))
             if skip_stale:
                 v_to.stale = False
         except ValueError as err:
             err_msg = getattr(err, 'message', str(err))
             var_val = value(v_from)
-            rounded_val = int(round(var_val))
+            rounded_val = round(var_val) if six.PY3 else int(round(var_val))
             # Check to see if this is just a tolerance issue
-            if 'is not in domain Binary' in err_msg and (
+            if ignore_integrality \
+                and ('is not in domain Binary' in err_msg
+                or 'is not in domain Integers' in err_msg):
+               v_to.value = value(v_from, exception=False)
+            elif 'is not in domain Binary' in err_msg and (
                     fabs(var_val - 1) <= config.integer_tolerance or
                     fabs(var_val) <= config.integer_tolerance):
                 v_to.set_value(rounded_val)
+            # TODO What about PositiveIntegers etc?
```

Comment:
If it's in a domain that is not explicitly handled here, then the user gets handed the exception via re-raise. If you would like to add in handling for the other domain groups, be my guest.

Reply:
I think somebody should have another look at it at some point, so I will keep the TODO tag.
But I won't work on it now.